src/window/message_dialog.c

```diff
@@ -118,18 +131,103 @@ static void set_city_message(int year, int month,
     player_message.use_popup = use_popup;
 }
 
-static void init(int text_id, void (*background_callback)(void))
+static void setup_custom_lang_message(int text_id)
+{
+    data.custom_lang_message.type = TYPE_MESSAGE;
+    data.custom_lang_message.message_type = MESSAGE_TYPE_CUSTOM;
+    data.custom_lang_message.x = 0;
+    data.custom_lang_message.y = 64;
+    data.custom_lang_message.width_blocks = 40;
+    data.custom_lang_message.height_blocks = 30;
+    data.custom_lang_message.title.x = 0;
+    data.custom_lang_message.title.y = 0;
+    data.custom_lang_message.subtitle.x = 16;
+    data.custom_lang_message.subtitle.y = 48;
+    data.custom_lang_message.urgent = 0;
+
+    data.custom_msg = custom_messages_get(text_id);
+    data.custom_lang_message.title.text = custom_messages_get_title(data.custom_msg);
+    data.custom_lang_message.subtitle.text = custom_messages_get_subtitle(data.custom_msg);
+    data.custom_lang_message.content.text = custom_messages_get_text(data.custom_msg);
+
+    if (custom_messages_get_video(data.custom_msg)) {
+        data.custom_lang_message.video.text = (uint8_t *) string_from_ascii(custom_messages_get_video(data.custom_msg));
+    }
+
+    data.should_play_audio = 0;
+    if (custom_messages_get_audio(data.custom_msg)) {
+        data.should_play_audio = 1;
+    }
+
+    data.should_play_background_music = 0;
+    if (custom_messages_get_background_music(data.custom_msg)) {
+        data.should_play_background_music = 1;
+    }
+}
+
+static void clear_custom_lang_message(void)
+{
+    data.custom_msg = 0;
+    data.custom_lang_message.title.text = 0;
+    data.custom_lang_message.subtitle.text = 0;
+    data.custom_lang_message.content.text = 0;
+    data.custom_lang_message.video.text = 0;
+    data.should_play_audio = 0;
+    data.should_play_background_music = 0;
+}
+
+static void init_audio(void)
+{
+    if (!data.show_video) {
+        if (data.should_play_audio) {
+            sound_speech_play_file(custom_messages_get_audio(data.custom_msg));
+        }
+        if (data.should_play_background_music) {
+            sound_device_stop_music();
+            int volume = setting_sound(SOUND_MUSIC)->volume;
+            sound_device_play_music(custom_messages_get_background_music(data.custom_msg), volume);
+        }
```

Comment:
In the future, I want a setting like `music_as_speech_background` that plays at a lower volume and fades out the music when the speech ends.

Reply:
I'm not sure how to do that.
I checked the existing windows that had to do with introductions (as far as I could tell) and didn't see anything like that.
Did I miss it somewhere?

Reply:
I completely misunderstood then when you first mentioned this in Discord.
I thought that was in code and already a feature, but I understand now you meant it was baked into the audio files.
When I didn't see it in code I assumed it is somehow baked into the `sound_device_play_music` code and I was just not seeing it.

I'll see if I can do a follow-up PR in the near future to include this feature.